docs/changes/2397.api.rst

```diff
@@ -0,0 +1,8 @@
+``CameraGeometry.position_to_pix_index`` will no return the minimum integer value for invalid
+pixel coordinates instead of -1 due to the danger of using -1 as an index in python accessing
+the last element of a data array for invalid pixels.
+The function will no also no longer raise an error if the arguments are empty arrays and instead
+also just return an empty index array.
```

Comment:
not sure if the "also" should be here, the function always returns one variable, not the the optional extra value implied by "also".